homeassistant/components/vizio/media_player.py

```diff
@@ -196,23 +208,39 @@ async def async_update(self) -> None:
             self._current_app = None
             self._current_app_config = None
             self._current_sound_mode = None
+            self._all_settings = {}
             return
 
         self._state = STATE_ON
 
-        audio_settings = await self._device.get_all_settings(
-            VIZIO_AUDIO_SETTINGS, log_api_exception=False
-        )
-        if audio_settings:
-            self._volume_level = float(audio_settings["volume"]) / self._max_volume
-            if "mute" in audio_settings:
-                self._is_volume_muted = audio_settings["mute"].lower() == VIZIO_MUTE_ON
+        if not self._setting_types:
+            self._setting_types = sorted(
+                await self._device.get_setting_types_list(log_api_exception=False)
+            )
+
+        for setting_type in self._setting_types:
+            self._all_settings[setting_type] = await self._device.get_all_settings(
```

Comment:
Are we doing this in sequence on purpose ? Why not parallel ?

Reply:
In testing I have found that the `vizio` API does not like concurrent requests, so I think it is safer to execute in sequence.